standards/SPIFFE_Workload_Endpoint.md

```diff
@@ -48,7 +48,7 @@ Clients may be explicitly configured with the socket location, or may utilize th
 
 The value of the `SPIFFE_ENDPOINT_SOCKET` environment variable is structured as an [RFC 3986](https://www.ietf.org/rfc/rfc3986.txt) URI. The scheme MUST be set to either `unix` or `tcp`, which indicates that the endpoint is served over a Unix Domain Socket or a TCP listen socket, respectively.
 
-If the scheme is set to `unix`, then the authority component MUST NOT be set, and the path component MUST be set to the absolute path of the SPIFFE Workload Endpoint Unix Domain Socket (e.g. `unix:///path/to/endpoint.sock`). The scheme and path components are mandatory, and no other component may be set.
+If the scheme is set to `unix`, then the authority component MUST NOT be set, and the path component MUST be set to the absolute path of the SPIFFE Workload Endpoint Unix Domain Socket (e.g. `unix:/path/to/endpoint.sock`). The scheme and path components are mandatory, and no other component may be set.
```

Comment:
Should an empty authority be allowed?  (i.e. should `unix:///path/` be acceptable?)

I feel that `unix:///path` and `unix:/path` are equivalent. RFC 3986 Section 3.2.2. provides an example of the `file` scheme:

> the "file" URI scheme is defined so that no authority, an empty host, and "localhost" all mean the end-user's machine, whereas the "http" scheme considers a missing authority or empty host invalid.

To toss out an alternative modification:

If the scheme is set to `unix`, then the authority component MUST NOT be set or, if the authority component is set, then the host must be empty.

Reply:
It seems that software consuming the URIs might not always handle `unix:///` correctly, if I'm understanding the comment on this issue correctly: https://github.com/parallaxsecond/parsec-client-rust/issues/37

Some deeper spelunking suggests that `unix` is not actually a registered scheme.  I can't find any authoritative document describing its semantics, and it's not listed in the IANA registry of schemes https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml

So... did we make this scheme up?

Reply:
I can't believe I've never noticed it wasn't an officially registered scheme. Probably because it's semantics match (maybe just superficially so) the `file` scheme sans the `host` component?

Reply:
I guess I had assumed that Parsec was implementing the SPIFFE Workload API, but on closer inspection it's not clear that's the case, and they're rolling their own wire protocol, but were clearly inspired by the SPIFFE spec for how to specify the endpoint.

Considering gRPC explicitly supports `unix:///path/to/endpoint.sock` URI syntax, I'm not sure it matters one way or another whether we merge this change.

Reply:
Yeah, as long as we're still requiring absolute path, I'm not sure what true benefit there is for allowing both forms.